http/typescript/fetch/src/dom.shim.d.ts

```diff
@@ -0,0 +1,14 @@
+/**
+ * -------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.
+ * See License in the project root for license information.
+ * -------------------------------------------------------------------------------------------
+ */
+
+interface Request { }
```

Comment:
No it doesn't. It gets merged if the definition is present. 

Example of  interface merging, 
```typescript
interface A{}

interface A{
    name: string
}

interface A {
    name: string,
    id: string
}

// We get a merged definition of A 
const a : A = {
    name: "",
    id: ""
}
```
This is a common approach used in isomorphic libraries to provide shims, that is, empty interfaces in case the user environment does not contain the DOM or any other required definitions.

TypeScript `interfaces` and `typings` do not transpile to JS. There is a lot of flexibility that the interfaces and types provide. 
[References and examples](https://github.com/microsoft/kiota/blob/af9a3c8e3e9cd84f42a47a1f6db32e6fb94266ac/http/typescript/fetch/docs/design/isomorphic.md#solution-to-this) in the  documented in the isomorphic readme .

Reply:
thanks for clarifying the point. Can you add a comment to set the context of why we have this in here for future reference please?

Reply:
thanks for clarifying the point. Can you add a comment to set the context of why we have this in here for future reference please?